### ISSUE TITLE
Assigned default containers to workspaces messed up on sync

### ISSUE THREAD

**Issue by sventurbo** [zen-browser/desktop#3429]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

> Device 1: Windows 11 (firefox sync enabled)
> Device 2: MacOS (firefox sync enabled)

I've got **multiple workspaces** assigned to **different containers** (e.g., workspace “work” with container “work” and so on…). When **syncing** both my devices, always the **same two workspaces switch their assigned container** (e.g., workspace “work” gets the “homelab” container assigned and the “homelab” workspace gets the “work” container assigned.)

I've already tried setting up a new 3. device (fresh OS install. Never used zen on it before) where this weird switch also occurs on sync.
The only **temporarily fix**, is to **always manually switch the containers** back to their workspaces after **every new start when it synced**.

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

Version 1.0.1-a.22 (64-bit)

### What platform are you seeing the problem on?

macOS - aarch64, Windows

**Comment by JSGRD22** [comment]
Use the container tabs extension for firefox to sync them properly.

**Comment by DCS-KC** [comment]
I have the same issue using two device, Windows 11 and Linux, and am using the Container Tabs extension.

**Comment by jerrwu** [comment]
I'm not sure if my issue is exactly the same. I have a default container set for one of my workspaces and when I sync, the default container setting can only exist on one device.

For example:

1. Set default container on desktop
2. Sync workspaces with laptop
3. Default container is no longer selected on laptop
4. Set it on laptop
5. It is no longer selected on desktop after a sync

**Comment by sventurbo** [comment]
> Use the container tabs extension for firefox to sync them properly.

Well I mentioned that I'm using firefox sync...

**Comment by A2va** [comment]
I have the same problem, I managed to get it to work by fiddling with the installation of Multi Account Containers. But since then I've reinstalled Zen and can't remember how to do it.

I suspect Multi Account Containers to be the root cause of this as it asks to log in to Firefox Sync even if the login was already made in the browser. I think the best way to fix this would be by implanting container sync like #2254 does.

**Comment by rsenna** [comment]
Yep, I’ve got the same issue on two separate macOS Sequoia machines. Whenever I log into the second machine, all “Default Containers” are replaced.

Interestingly, the replacement isn’t random: it consistently switches to the same (wrong) container.

Only workaround for now seems to be disabling workspace sync entirely. Bummer.

**Comment by emmyapi** [comment]
I have this issue too. I have 6 work spaces with 6 different containers. The defaults are all mixed up every time I switch device

**Comment by mr-cheffy** [comment]
## Closing this issue as part of a repo cleanup

Due to technical-debt we had in the past, we are currently clearing out old, duplicate, stale, and non-actionable issues to improve issue tracking and make it easier to focus on fixing real bugs.

If this issue is still relevant, please open a new issue using our updated issue template, which will help us categorize and address it more efficiently.

Thanks for your understanding and for helping us improve the project! 🚀

This issue can be reopened at: https://github.com/zen-browser/desktop/issues/new?template=bug_report.yml&title=Assigned%20default%20containers%20to%20workspaces%20messed%20up%20on%20sync%0A
